closed/make/Javadoc.gmk

```diff
@@ -0,0 +1,486 @@
+	$(prep-target)
+	@($(call OptionOnly,$(COMMON_JAVADOCFLAGS)) ; \
+	  $(call OptionPair,-sourcepath,$(OPENJ9_DOCS_SRCDIR)) ; \
+	  $(call OptionPair,-encoding,ascii) ; \
+	  $(call OptionPair,-doctitle,$(OPENJ9_DATAACCESS_DOCTITLE)) ; \
+	  $(call OptionPair,-windowtitle,$(OPENJ9_DATAACCESS_WINDOWTITLE) $(DRAFT_WINTITLE)) ; \
+	  $(call OptionPair,-header,$(OPENJ9_DATAACCESS_HEADER)$(DRAFT_HEADER)) ; \
+	  $(call OptionPair,-bottom,$(OPENJ9_JAVADOC_BOTTOM)$(DRAFT_BOTTOM)) ; \
+	  $(call OptionTrip,-linkoffline,$(JAVASE_BASE_URL),$(COREAPI_DOCSDIR)/) ; \
+	) > $@
+
+# Create a file with the package names in it
+$(OPENJ9_DATAACCESS_PACKAGES_FILE): $(DIRECTORY_CACHE) $(call PackageDependencies,$(OPENJ9_DATAACCESS_PKGS))
+	$(prep-target)
+	$(call PackageFilter,$(OPENJ9_DATAACCESS_PKGS))
+
+#############################################################################
+#
+# openj9-jvmdocs
+#
+
+OPENJ9_JVM_PKGS := com.ibm.jvm
+
+OPENJ9_JVM_FIRST_COPYRIGHT_YEAR := 1998
+
+OPENJ9_JVM_DOCDIR := $(PLATFORM_DOCSDIR)/jvm
+OPENJ9_JVM2COREAPI := $(JDKJRE2COREAPI)
+OPENJ9_JVM_DOCTITLE := OpenJ9 JVM Diagnostic Utilities Classes
+OPENJ9_JVM_WINDOWTITLE := OpenJ9 JVM Diagnostic Utilities Classes
+OPENJ9_JVM_HEADER := <strong>OpenJ9 JVM Diagnostic Utilities Classes</strong>
+
+# The index.html, options, and packages files
+OPENJ9_JVM_INDEX_FILE := $(OPENJ9_JVM_DOCDIR)/index.html
+OPENJ9_JVM_OPTIONS_FILE := $(DOCSTMPDIR)/openj9.jvm.options
+OPENJ9_JVM_PACKAGES_FILE := $(DOCSTMPDIR)/openj9.jvm.packages
+
+openj9-jvmdocs: $(OPENJ9_JVM_INDEX_FILE)
+
+# Set relative location to platform document root
+$(OPENJ9_JVM_INDEX_FILE): GET2DOCSDIR := $(PLATFORM_DOCSDIR)/..
+
+# Run javadoc if the index file is out of date or missing
+$(OPENJ9_JVM_INDEX_FILE): $(OPENJ9_JVM_OPTIONS_FILE) $(OPENJ9_JVM_PACKAGES_FILE)
+	$(prep-javadoc)
+	$(call JavadocSummary,$(OPENJ9_JVM_OPTIONS_FILE),$(OPENJ9_JVM_PACKAGES_FILE))
+	$(JAVADOC_CMD) -d $(@D) \
+	    @$(OPENJ9_JVM_OPTIONS_FILE) @$(OPENJ9_JVM_PACKAGES_FILE)
+
+$(OPENJ9_JVM_OPTIONS_FILE): FIRST_COPYRIGHT_YEAR := $(OPENJ9_JVM_FIRST_COPYRIGHT_YEAR)
+$(OPENJ9_JVM_OPTIONS_FILE):
+	$(prep-target)
+	@($(call OptionOnly,$(COMMON_JAVADOCFLAGS)) ; \
+	  $(call OptionPair,-sourcepath,$(OPENJ9_DOCS_SRCDIR)) ; \
+	  $(call OptionPair,-encoding,ascii) ; \
+	  $(call OptionPair,-doctitle,$(OPENJ9_JVM_DOCTITLE)) ; \
+	  $(call OptionPair,-windowtitle,$(OPENJ9_JVM_WINDOWTITLE) $(DRAFT_WINTITLE)) ; \
+	  $(call OptionPair,-header,$(OPENJ9_JVM_HEADER)$(DRAFT_HEADER)) ; \
+	  $(call OptionPair,-bottom,$(OPENJ9_JAVADOC_BOTTOM)$(DRAFT_BOTTOM)) ; \
+	  $(call OptionTrip,-linkoffline,$(JAVASE_BASE_URL),$(COREAPI_DOCSDIR)/) ; \
+	) > $@
+
+# Create a file with the package names in it
+$(OPENJ9_JVM_PACKAGES_FILE): $(DIRECTORY_CACHE) $(call PackageDependencies,$(OPENJ9_JVM_PKGS))
+	$(prep-target)
+	$(call PackageFilter,$(OPENJ9_JVM_PKGS))
+
+#############################################################################
+#
+# openj9 mgmtdocs
```

Comment:
Mostly a curiosity: why did you replace the hyphen with a space here (all similar comments still use a hyphen)?

Reply:
In the previous comments (with a hyphen) the comment is also the name of the target which is run.
As a result of removing the phony dependencies in openj9 jdk management docs targets this is not now the case in this section (there is no longer an openj9-mgmtdocs target).